packages/expect-utils/src/utils.ts

```diff
@@ -137,7 +137,7 @@ export const getObjectSubset = (
 const IteratorSymbol = Symbol.iterator;
 
 const hasIterator = (object: any) =>
-  !!(object != null && object[IteratorSymbol]);
+  Boolean(object != null && object[IteratorSymbol]);
```

Comment:
`object != null` already yields boolean, so it is wrong
yoy want something like `object != null && Boolean(object[IteratorSymbol]);`

Reply:
I think we should do `object != null && object[IteratorSymbol] != null`

Reply:
yeah i am sort of confused but this looks good